CODING_STYLE.md

```diff
@@ -0,0 +1,590 @@
+   standard (pre 2024), the `export`/`readonly` built-in may convert the
+   assignment to behave like a command under some circumstances (like parts of
+   the arguments being quoted) and suddenly field splitting will be performed
+   again, unlike for all other assignments ([Austin Group Defect 351]).
+
+   To prevent such "surprises" assigning to a variable and exporting it or
+   marking it as readonly shall always be done in two steps.
+
+       # Don't
+       export foo=$(echo foo bar; false)    # exit status 0
+       echo "${foo}"                        # foo bar
+       export 'foo'=$(echo foo bar; false)  # exit status 0
+       echo "${foo}"                        # foo
+       'ex'port foo=$(echo foo bar; false)  # exit status 0
+       echo "${foo}"                        # foo
+
+       # Do
+       foo=$(echo foo bar; false)           # exit status 1
+       export foo                           # exit status 0
+       echo "${foo}"                        # foo bar
+
+4. Parameter expansions (use of a variable)
+
+   **Parameter expansions always use braces.**
+
+   It makes searching for usages of a variable easier because it allows for
+   `grep -F '${my_var'` to be used without having to differenciate between
+   `"$my_var"`, `"${my_var:-default}"`, etc.
+   The only exception being usage in arithmetic expansions.
+
+   In addition, expansion of skonfig-provided environment variables
+   (e.g. `__object`, `__global`, etc.) always use the "indicate error if null or
+   unset" operators `${:?}`.  
+   This helps to reduce the likelyhood of users inadvertently executing skonfig
+   scripts in their regular shell, because the shell will terminate with a
+   variable unset error.  
+   Further, it "resolves" ShellCheck [SC2154] errors.
+
+       # Don't
+       echo "$x"
+       stdin_file="$__object/stdin"
+
+       # Do
+       echo "${x}"
+       stdin_file="${__object:?}/stdin"
+
+4. Clean up temporarily used variables after yourself (`unset -v`).
+
+   There are only few things more annoying than to debug errors which occur due to
+   some old variable left lying around and having a random value triggering
+   unexpected behaviour.
+
+   Also, the behaviour of code should never change because of some random state
+   of external variables.
+   Always initialise variables before use or `unset -v` them before use when in
+   doubt ([POSIX.1-2024 unset]).
+
+5. Do not quote command substitutions when assigning to a variable.
+
+   No field splitting is done by the shell when assigning to a variable
+   ([POSIX.1-2024 2.9.1.1], 4.).  
+   Quoting will only unnecessarily disable syntax highlighting in most editors.
+
+       # Don't
+       myos="$(uname -s)"
+
+       # Do
+       myos=$(uname -s)
+
+
+## 9. Common commands
+
+1. grep
+
+   1. Always use fixed string grep (`grep -F`) when grepping for a literal
+      string or a parameter expansion which does not contain a regular
+      expression.
```

Comment:
I'd expand this with `-x`.

If you need to check whole line, then instead of `grep '^...$'`, use `grep -Fx`. 

Maybe in separate bullet in this section?